tools/dart/dart_roll_helper.py

```diff
@@ -0,0 +1,208 @@
+    'out/android_debug',
+    'out/android_profile_unopt',
+    'out/android_profile',
+    'out/android_release_unopt',
+    'out/android_release'
+  ]
+  for config in configs:
+    p = subprocess.Popen(command + ['-C', config], cwd=ENGINE_HOME)
+    p.wait()
+
+def run_tests():
+  print_status('Running tests in packages/flutter')
+  p = subprocess.Popen(FLUTTER_TEST + ['--local-engine=host_debug'],
+                       cwd=PACKAGE_FLUTTER)
+  p.wait()
+  print_status('Running tests in examples/flutter_gallery')
+  p = subprocess.Popen(FLUTTER_TEST + ['--local-engine=host_debug'],
+                       cwd=FLUTTER_GALLERY);
+  p.wait()
+
+def run_hot_reload_configurations():
+  print_status('Running flutter gallery release')
+  p = subprocess.Popen(FLUTTER_RUN + ['--release', '--local-engine=android_release'],
+                       cwd=FLUTTER_GALLERY)
+  p.wait()
+  print_status('Running flutter gallery debug')
+  p = subprocess.Popen(FLUTTER_RUN + ['--local-engine=android_debug_unopt'],
+                       cwd=FLUTTER_GALLERY)
+  p.wait()
+
+def update_licenses():
+  print_status('Updating Flutter licenses')
+  p = subprocess.Popen([ENGINE_LICENSE_SCRIPT], cwd=ENGINE_HOME)
+  p.wait()
+  LICENSE_SCRIPT_OUTPUT = 'out/license_script_output/'
+  src_files = os.listdir(LICENSE_SCRIPT_OUTPUT)
+  for f in src_files:
+    path = os.path.join(LICENSE_SCRIPT_OUTPUT, f)
+    if os.path.isfile(path):
+      shutil.copy(path, ENGINE_GOLDEN_LICENSES)
+
+def get_commit_range(start, finish):
+  range_str = '{}...{}'.format(start, finish)
+  command = ['git', 'log', '--oneline', range_str]
+  orig_dir = os.getcwd()
+  os.chdir(DART_SDK_HOME)
+  result = subprocess.check_output(command)
+  os.chdir(orig_dir)
+  return result
+
+def git_commit():
+  global original_revision
+  global updated_revision
+
+  print_status('Committing Dart SDK roll')
+  ENGINE_FLUTTER = ENGINE_HOME + 'flutter'
+  current_date = datetime.date.today()
+  sdk_log = get_commit_range(original_revision, updated_revision)
+  commit_msg = 'Dart SDK roll for {}\n\n'.format(current_date)
+  commit_msg += sdk_log
+  commit_cmd = ['git', 'commit', '-a', '-m', commit_msg]
+  p = subprocess.Popen(commit_cmd, cwd=ENGINE_FLUTTER)
+  p.wait()
+
+def main():
+  global updated_revision
+
+  parser = argparse.ArgumentParser(description='Automate most Dart SDK roll tasks.')
+  parser.add_argument('dart_sdk_revision', help='Target Dart SDK revision')
+  parser.add_argument('--create_commit', action='store_true',
+                      help='Create the engine commit with Dart SDK commit log')
+  parser.add_argument('--no_test', action='store_true',
+                      help='Skip running tests and hot reload configurations')
+  parser.add_argument('--no_update_licenses', action='store_true',
+                      help='Skip updating licenses')
+  args = parser.parse_args()
+  updated_revision = args.dart_sdk_revision
+
+  print_status('Starting Dart SDK roll')
```

Comment:
It might be good to have flags that let you pick up where you left off if a step fails.